### PR TITLE
[R4R] BEP3 module spec and clean up

### PR DESCRIPTION
**Pull request by denalimarsh**
Closes issues: https://github.com/Kava-Labs/kava/issues/384, https://github.com/Kava-Labs/kava/issues/425

Primary changes:
- Added module spec
- Added 'swaps_expired' event
- Updated event emission to always be after state updates
- Removed `Swap` interface

~~This PR is **blocked** until https://github.com/Kava-Labs/kava/pull/421 is merged. Once https://github.com/Kava-Labs/kava/pull/421 is merged, the base branch will be updated to master.~~

The PR's base branch has been set to master and is ready to merge.

### CODE REVIEW

File: x/bep3/spec/02_state.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+## Parameters and genesis state
+
+`Paramaters` define the rules according to which swaps are executed. Parameter updates can be made via on-chain parameter update proposals.
+
+```go
+// Params governance parameters for bep3 module
+type Params struct {
+	BnbDeputyAddress sdk.AccAddress `json:"bnb_deputy_address" yaml:"bnb_deputy_address"`
+	MinBlockLock     int64          `json:"min_block_lock" yaml:"min_block_lock"`
```
Comment (karzak) [inline x/bep3/spec/02_state.md:11]: I would add a comment to block-lock params that they represent block height span, not time or some other measure.

File: x/bep3/spec/01_concepts.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Concepts
+
+ BEP3 is a protocol for securely transferring assets between Kava and chains that implement the BEP3 protocol via Atomic Swaps such as Binance Chain. Tranactions are witnessed and relayed between the two blockchains by Binance's BEP3 deputy process. The deputy maintains an address on both chains and is responsible for delivering tokens upon the successful completion of an Atomic Swap. Learn more about the BEP3 deputy process [here](https://github.com/binance-chain/bep3-deputy).
```
Comment (karzak) [inline x/bep3/spec/01_concepts.md:3]: I would link to original Binance Chain BEP3 doc: https://github.com/binance-chain/BEPs/blob/master/BEP3.md

File: x/bep3/spec/03_messages.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	SwapID       tmbytes.HexBytes `json:"swap_id"  yaml:"swap_id"`
+	RandomNumber tmbytes.HexBytes `json:"random_number"  yaml:"random_number"`
+}
+```
+
+
+
+## Refund swap
+
+Swaps are claimed using the `MsgRefundAtomicSwap` message type.
```
Comment (karzak) [inline x/bep3/spec/03_messages.md:40]:
```suggestion
Expired swaps are refunded using the `MsgRefundAtomicSwap` message type.
```

Comment (denalimarsh) [reply x/bep3/spec/03_messages.md]: https://github.com/Kava-Labs/kava/pull/450/commits/2516b8d15cef7d1fa013aaef9ddb75171e7ad413

File: x/bep3/spec/01_concepts.md (hunk 2 above)
Comment (denalimarsh) [reply x/bep3/spec/01_concepts.md]: https://github.com/Kava-Labs/kava/pull/450/commits/2516b8d15cef7d1fa013aaef9ddb75171e7ad413

File: x/bep3/spec/02_state.md (hunk 1 above)
Comment (denalimarsh) [reply x/bep3/spec/02_state.md]: https://github.com/Kava-Labs/kava/pull/450/commits/2516b8d15cef7d1fa013aaef9ddb75171e7ad413

**Comment by denalimarsh** [comment]
This PR is **blocked** and should not be merged at this time.

File: x/bep3/keeper/swap.go [hunk 4]
```diff
@@ -233,11 +236,21 @@ func (k Keeper) UpdateExpiredAtomicSwaps(ctx sdk.Context) error {
...
+		k.RemoveFromByBlockIndex(ctx, atomicSwap)
+
+		// Emit 'swap_expired' event
+		ctx.EventManager().EmitEvent(
+			sdk.NewEvent(
+				types.EventTypeSwapExpired,
+				sdk.NewAttribute(types.AttributeKeyAtomicSwapID, fmt.Sprintf("%s", hex.EncodeToString(atomicSwap.GetSwapID()))),
+				sdk.NewAttribute(types.AttributeExpirationBlock, fmt.Sprintf("%d", ctx.BlockHeight())),
+			),
+		)
```
Comment (fedekunze) [inline x/bep3/keeper/swap.go:251]: you can add all the swapIDs to an attribute array and then emit a single event

Comment (denalimarsh) [reply x/bep3/keeper/swap.go]: https://github.com/Kava-Labs/kava/pull/450/commits/94cc38c3d22bae9a6178174be46bc0ef128797d9

**Pull request merged**